### PR TITLE
fix: #32 map not match a correct type

### LINKED ISSUE #32: Types mismatch when using `map` operator
When `map` converts source observable type to a different one, the program does not compile.

This is a simple error reproducer, can be added to the `map.rs' file in unit tests section:

```rust
  #[test]
  fn map_types_mixed() {
    let mut i = 0;
    observable::from_iter!(vec!['a','b','c'])
      .map(|v| 1)
      .subscribe(|v| i += *v);
    assert_eq!(i, 3);
  }
```

The compilation error received is:
```
   Compiling rxrust v0.4.1-alpha.0 (/workspaces/rxRust)
error[E0277]: cannot add-assign `char` to `{integer}`
   --> src/ops/map.rs:223:24
    |
223 |       .subscribe(|v| i += *v);
    |                        ^^ no implementation for `{integer} += char`
    |
    = help: the trait `std::ops::AddAssign<char>` is not implemented for `{integer}`

error[E0631]: type mismatch in closure arguments
   --> src/observable/from.rs:7:31
    |
7   |     let _: &Observer<_, ()> = &subscriber;
    |                               ^^^^^^^^^^^ expected signature of `for<'r> fn(&'r {integer}) -> _`
    | 
   ::: src/ops/map.rs:221:5
    |
221 |     observable::from_iter!(vec!['a','b','c'])
    |     ----------------------------------------- in this macro invocation
222 |       .map(|v| 1)
223 |       .subscribe(|v| i += *v);
    |                  ----------- found signature of `for<'r> fn(&'r char) -> _`
    |
    = note: required because of the requirements on the impl of `subscribable::Observer<{integer}, ()>` for `subscribable::subscribable_pure::SubscribePure<[closure@src/ops/map.rs:223:18: 223:29 i:_]>`
    = note: required because of the requirements on the impl of `subscribable::Observer<char, ()>` for `ops::map::MapSubscribe<subscribable::subscribable_pure::SubscribePure<[closure@src/ops/map.rs:223:18: 223:29 i:_]>, [closure@src/ops/map.rs:222:12: 222:17]>`
    = note: required because of the requirements on the impl of `subscribable::Observer<char, ()>` for `subscriber::Subscriber<ops::map::MapSubscribe<subscribable::subscribable_pure::SubscribePure<[closure@src/ops/map.rs:223:18: 223:29 i:_]>, [closure@src/ops/map.rs:222:12: 222:17]>, subscription::LocalSubscription>`
    = note: required for the cast to the object type `dyn subscribable::Observer<char, ()>`

```

### PR DESCRIPTION
**Pull request by M-Adoo**

### CODE REVIEW

**Comment by M-Adoo** [comment]
@achary can you help me to review the code for fix #32 

And BTW, do you have any interesting to maintain this project together ? We can take turn to release and announce version,  also review code by each other.   More important it we can discuss and do more to promote the development of this project.

File: src/ops/map.rs [hunk 1]
```diff
@@ -56,12 +60,12 @@ where
   }
 }
 
-pub struct MapSubscribe<S, M> {
+pub struct MapObserver<S, M> {
```
Comment (achary) [inline src/ops/map.rs]: Thanks for this rename. It made it much clearer what the role is and how it fits in the overall jigsaw puzzles.

**Comment by achary** [comment]
WRT greater involvement to `rxrust`. I will be surely trying to help as much as I can.

There are few aspects:
- reviewing anyones fixes for issues I've raised. This is something you can count always count on :) so, yes, please don't hesitate here.
- fixes for issues I've raised - I will try to be also able to fix things if I see the way to do it. This should improve over time, as the very deep guts of rxrust are still not entirely transparent to me at the very now.
- co-maintanence of the project. Here, the limitation on my end is that time I could spend on it is rather short and comes in random  bursts of availability (few days here and there) and then potential longer (few weeks, month) pauses of not being able to jump on issues quickly or at all.
My take would be - let's see how it develops for let's say next two months and then we could have a better understanding of how it could work and whether it would be good the project, OK?

**Comment by M-Adoo** [comment]
OK，for this pr, please approve it. let's have a good start.
image

**Comment by achary** [comment]
Hmm. I think I've already approved it. The only way I have available is over Add your review option, there I can select approve. But that doesn't seem to change anything here.
Is that a permission thing?

**Comment by M-Adoo** [comment]
I add you to collaborators let you have write access to this project.  maybe changed after you accept the invite.

**Comment by M-Adoo** [comment]
https://github.com/M-Adoo/rxRust/invitations

**Comment by achary** [comment]
Thanks, all worked now.

**Pull request merged**